### PR TITLE
Feat/product swap

### LINKED ISSUE #49: Let user swap products
## Value Statement

As a user, when I try to add a product to e.g. monday morning, that has conflicts with a product that is already added to monday morning, I want to be given the opportunity to swap the current product with the conflicting product in my routine., so that i don't have to go to the conflicting product to edit the times manually and then come back to the current to add it.

## Acceptance Criteria

* User needs to be alerted when trying to add conflicting products
* User needs to be given two buttons: cancel and swap
* Clicking swap will remove the conflicting product and add the current product instead

## Tasks

- [x] Create feature branch
- [x] Implement logic to remove conflicting product on "swap" clicked
- [x] Implement logic to send new current product info to LS

## Complexity
medium

### PR DESCRIPTION
**Pull request by FriederikeM**
* Let user swap conflicting products 
* Don't let user add conflicting together anymore

### CODE REVIEW

File: src/components/FormModal/AlertModal.js [hunk 1]
```diff
@@ -2,25 +2,19 @@ export default function AlertModal({
   conflictName,
   clickedWeekdayName,
   clickedTimeOfTheDay,
-  onAddAnywayClicked,
+  onProductSwapClicked,
```
Comment (hugh-burgess) [inline src/components/FormModal/AlertModal.js:7]: Incredible renaming

File: src/components/FormModal/AlertModal.js [hunk 2]
```diff
@@ -2,25 +2,19 @@ export default function AlertModal({
...
-        </em>{" "}
-        on {clickedWeekdayName} {clickedTimeOfTheDay}. These products have
-        conflicting ingredients
+        You are already using <em>{conflictName}</em> on {clickedWeekdayName}{" "}
+        {clickedTimeOfTheDay}. These products have conflicting ingredients.
+        Would you like to swap these two products?
       </p>
       <div className="alert-modal-finishing-button-wrapper">
-        <button className="add-anyway" onClick={onAddAnywayClicked}>
-          add anyway
```
Comment (hugh-burgess) [inline src/components/FormModal/AlertModal.js:23]: Class name “add anyway”? Come on let’s have a more specific one

File: src/components/FormModal/FormModal.js [hunk 3]
```diff
@@ -166,7 +169,10 @@ export default function FormModal({
       alert(
         "Please specify during which time of the day you want to use the product on the checked days"
       );
-    } else if (isNothingChecked === true) {
+    } else if (isNothingChecked === true && editMode === false) {
+      onCancelAdding();
+    } else if (isNothingChecked === true && editMode === true) {
```
Comment (hugh-burgess) [inline src/components/FormModal/FormModal.js:174]: Couldn’t these just be `if`?

File: src/utility/findConflictingProduct.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import getIndexForWeekday from "./getIndexForWeekday";
 import getProductById from "./getProductById";
 
-export function findConflictingProductIds(
+export function findConflictingProductId(
```
Comment (hugh-burgess) [inline src/utility/findConflictingProduct.js:4]: Incredible effort today

File: src/utility/localStorage.js [hunk 5]
```diff
@@ -19,3 +19,13 @@ export function editDataInLocalStorage(items) {
 
   localStorage.setItem("routine", JSON.stringify(newData));
 }
+
+export default function removeProductFromLocalStorage(bla) {
```
Comment (hugh-burgess) [inline src/utility/localStorage.js]: Hahaha “bla”. Think about future Rikki

**Review by hugh-burgess — CHANGES_REQUESTED** [review]
Great PR for the lads. Lgtmtmttgm

File: src/utility/localStorage.js (hunk 5 above)
Comment (FriederikeM) [reply src/utility/localStorage.js]: woops... 🙈

**Review by soerendolling — APPROVED** [review]
looks all fine to me

File: src/components/FormModal/AlertModal.css [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  text-align: center;
+  height: auto;
+  width: 80vw;
+  border-radius: 15px;
+  border: 1px solid var(--color-three-opaque);
+  padding: 20px 10px 5px;
+  position: fixed;
+  top: 18vh;
+  left: 5vw;
+  filter: none important!;
```
Comment (marwinburesch) [inline src/components/FormModal/AlertModal.css]: It is used a little differently, if at all :) 
https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#the_!important_exception

File: src/components/FormModal/AlertModal.css [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  color: var(--color-one);
+  background: linear-gradient(
+    145deg,
+    var(--color-three-opaque),
+    var(--color-three)
+  );
+  backdrop-filter: blur(10px);
+  z-index: 100;
+  text-align: center;
+  height: auto;
```
Comment (marwinburesch) [inline src/components/FormModal/AlertModal.css]: Should be default

File: src/components/FormModal/AlertModal.css [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+  width: 80vw;
```
Comment (marwinburesch) [inline src/components/FormModal/AlertModal.css]: I think you're overwriting this with a media query, which doesn't do much (I think). You could just use the `left` and `right` properties to assign a width. Or leave it at 80% always 🤷

File: src/components/FormModal/AlertModal.css [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+  display: flex;
+  flex-direction: column;
+  justify-content: space-around;
+  gap: 30px;
+  animation: alert-pop-up 800ms ease-in-out;
+}
+
+@keyframes alert-pop-up {
+  from {
+    position: absolute;
```
Comment (marwinburesch) [inline src/components/FormModal/AlertModal.css]: Isn't it fixed?

File: src/components/FormModal/AlertModal.css [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+.alert-message {
+  line-height: 1.3rem;
+}
+
+.alert-modal-finishing-button-wrapper > button {
+  border-radius: 8px;
+  font-weight: bold;
+  border: none;
+  width: 100px;
+  height: 40px;
```
Comment (marwinburesch) [inline src/components/FormModal/AlertModal.css:55]: 🤔

**Review by marwinburesch — CHANGES_REQUESTED** [review]
I think you've increased readability, excellent! 🐜 
Good, risky call on this feature. It's done 🎉 

`src/components/FormModal/Form.css` doesn't have to be in this PR :)

**Pull request merged**